src/Microsoft.OpenApi/Writers/OpenApiWriterExtensions.cs

```diff
@@ -137,7 +138,7 @@ public static void WriteProperty<T>(this IOpenApiWriter writer, string name, T v
         public static void WriteOptionalObject<T>(
             this IOpenApiWriter writer,
             string name,
-            T value,
+            T? value,
```

Comment:
shouldn't this be replicated for the collection method?

Reply:
Yes, but for now I've used #nullable enable/#nullable restore block to limit the amount of changes here because it's not required in this PR